src/main/java/net/mcreator/element/types/Block.java

```diff
@@ -356,4 +355,76 @@ private Image getTextureWithFallback(String textureName) {
 		return baseTypes;
 	}
 
+	@Override public Collection<String> getUsedElementNames() {
+		List<String> elements = new ArrayList<>(spawnWorldTypes);
+		if (guiBoundTo != null && !guiBoundTo.equals("<NONE>"))
+			elements.add("CUSTOM:" + guiBoundTo);
+		return elements;
+	}
+
+	@Override public Collection<? extends MappableElement> getUsedElementMappings() {
+		List<MappableElement> elements = new ArrayList<>();
+		elements.add(creativePickItem);
+		elements.add(soundOnStep);
+		if (!creativeTab.getUnmappedValue().equals("No creative tab entry"))
+			elements.add(creativeTab);
+		if (!useLootTableForDrops)
+			elements.add(customDrop);
+		if (hasInventory)
+			elements.addAll(fluidRestrictions);
+		if (doesGenerateInWorld()) {
+			elements.addAll(restrictionBiomes);
+			elements.addAll(blocksToReplace);
+		}
+		return elements;
+	}
+
+	@Override public Collection<? extends Procedure> getUsedProcedures() {
+		List<Procedure> procedures = new ArrayList<>();
+		procedures.add(placingCondition);

```

Comment:
This system is intended to report used procedures, and "" does not sound like a used procedure.

You can make helper methods in interfaces to filter lists, for example, but is is not acceptable to allow for additional data that does not describe anything and is only there because we couldn't do better

Reply:
I'll go with the ugly solution first.

Reply:
✔️ Fixed (?)